keps/sig-cloud-provider/2436-controller-manager-leader-migration/README.md

```diff
@@ -349,18 +354,131 @@ feature without providing a configuration, the default configuration will reflec
 Yes. Once the feature is enabled via feature gate, it can be disabled by unsetting `--enable-leader-migration` on KCM
 and CCM.
 
+###### What happens if we reenable the feature if it was previously rolled back?
+
+This feature can be re-enabled without side effects.
+
 ###### Are there any tests for feature enablement/disablement?
 
 Yes. Unit & integration tests include flag/configuration parsing. E2E test will have cases with the feature enabled and
 disabled.
 
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The rollout may fail if the configuration file does not represent correct controller-to-manager.
+This can cause controllers referred in the configuration file to either be unavailable or run in multiple instances.
+
+The rollback may fail if the leader election of the controller manager is not properly configured.
+For example, multiple instances of the same controller manager are running without election, or none of the instances become the leader.
+In these situations, all controllers will be either unavailable or conflict among multiple instances.
+
+###### What specific metrics should inform a rollback?
+
+If neither controller managers show `leader_active` for the main leader lock or the migration lock, Leader Migration may fail to activate and thus needs rollback.
+If any of the controllers indicate they are unavailable through their per-controller metric, Leader Migration may need reconfiguration.
+The metrics of each controller are specific to the implementation of each cloud provider and out of scope of this KEP,
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+The manual testing showed a clean takeover of the migration leader during both upgrade and downgrade process.
+This process will be tested as part of the e2e suite, required by [Graduation Criteria](#graduation-criteria).
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+None.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+N/A. This feature is never used by any user workloads.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] Other (treat as last resort)
+  - Details: The cluster administrator SHOULD have access to logs and metrics during cluster upgrade.
```

Comment:
The migration lock will be visible as a resource.

Reply:
Good point. And the leader of such lease would be visible too.